test/minikube/minikube_fsfreeze_test.go

```diff
@@ -0,0 +1,228 @@
+	admin := fmt.Sprintf("%s-nuodb-cluster0", helmChartReleaseName)
+	admin0 := fmt.Sprintf("%s-0", admin)
+	defer testlib.Teardown(testlib.TEARDOWN_DATABASE)
+	sourceDatabaseChartName := testlib.StartDatabase(t, namespaceName, admin0, &helm.Options{
+		SetValues: map[string]string{
+			"database.sm.resources.requests.cpu":                         testlib.MINIMAL_VIABLE_ENGINE_CPU,
+			"database.sm.resources.requests.memory":                      testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"database.te.resources.requests.cpu":                         testlib.MINIMAL_VIABLE_ENGINE_CPU,
+			"database.te.resources.requests.memory":                      testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"database.persistence.storageClass":                          testlib.SNAPSHOTABLE_STORAGE_CLASS,
+			"database.sm.noHotCopy.journalPath.persistence.storageClass": testlib.SNAPSHOTABLE_STORAGE_CLASS,
+			"database.sm.noHotCopy.journalPath.enabled":                  "true",
+			"database.sm.noHotCopy.replicas":                             "1",
+			"database.sm.hotCopy.replicas":                               "0",
+			"database.backupHooks.enabled":                               "true",
+		},
+	})
+
+	// Invoke backup hooks on SM pod
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
+	smPod := fmt.Sprintf("sm-%s-nuodb-cluster0-demo-0", sourceDatabaseChartName)
+	backupId := "123abc"
+	t.Run("verifyWritesFrozen", func(t *testing.T) {
+		// Freeze writes to archive and defer invocation of unfreeze
+		testlib.InvokeBackupHook(t, namespaceName, smPod, "pre-backup/"+backupId)
+		defer testlib.InvokeBackupHook(t, namespaceName, smPod, "post-backup/"+backupId)
+
+		// Try to create a file in archive directory in the background
+		k8s.RunKubectl(t, kubectlOptions, "exec", smPod, "-c", "engine", "--",
+			"sh", "-c", "nohup touch /var/opt/nuodb/archive/test-xyz &")
+
+		// Wait some time and check that file has not been created
+		time.Sleep(3 * time.Second)
+		output, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", smPod, "-c", "engine", "--",
+			"ls", "/var/opt/nuodb/archive")
+		require.NoError(t, err, output)
+		require.NotContains(t, output, "test-xyz")
+	})
+
+	// Wait short amount of time for test file to be created
+	testlib.Await(t, func() bool {
+		output, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", smPod, "-c", "engine", "--",
+			"ls", "/var/opt/nuodb/archive")
+		require.NoError(t, err, output)
+		return strings.Contains(output, "test-xyz")
+	}, 50*time.Millisecond)
+
+	// Check that backup hook sidecar logged expected messages
+	output, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "logs", smPod, "-c", "backup-hooks")
+	require.NoError(t, err, output)
+	require.Contains(t, output, "Freezing writes to archive volume")
+	require.Contains(t, output, "Unfreezing writes to archive volume")
+
+	t.Run("negativeTests", func(t *testing.T) {
+		// Negative test: use a bogus resource path
```

Comment:
Can you also add a negative test with multiple path components (for example `/pre-backup/backupId/bogus`)?